apps/dashboard/src/@/components/ui/DynamicHeight.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+
+import type React from "react";
+import { useLayoutEffect, useRef, useState } from "react";
+
+export function DynamicHeight(props: {
+  children: React.ReactNode;
+  maxHeight?: string;
+  transition?: string;
+}) {
+  const { height, elementRef } = useHeightObserver();
+
+  return (
+    <div
+      style={{
+        height: height ? `${height}px` : "auto",
+        transition:
+          props.transition ||
+          "height 210ms cubic-bezier(0.175, 0.885, 0.32, 1.1)",
+        overflow: "hidden",
+        boxSizing: "border-box",
+      }}
+    >
+      <div
+        ref={elementRef}
+        style={{
+          maxHeight: props.maxHeight,
+        }}
+      >
+        {props.children}
+      </div>
+    </div>
+  );
+}
+
+export function useHeightObserver() {
+  const elementRef = useRef<HTMLDivElement>(null);
+  const [height, setHeight] = useState<number | undefined>();
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: ok
```

Comment:
what was it complaining about here?

Reply:
Having extra dependencies, removing it - don't need it